tests/integration/devfile/utils/utils.go

```diff
@@ -186,3 +189,81 @@ func ExecWithRestartAttribute(projectDirPath, cmpName, namespace string) {
 	Expect(output).To(ContainSubstring("if not running"))
 
 }
+
+// OdoWatch creates files, dir in the context and watches for the changes to be pushed
+// this was taken from e2e_images_test.go and modified for devfile
+func OdoWatch(cmpName, project, context string, stringsToBeMatched []string, runner interface{}, platform string) {
+
+	startSimulationCh := make(chan bool)
+	go func() {
+		startMsg := <-startSimulationCh
+		if startMsg {
+			err := os.MkdirAll(filepath.Join(context, ".abc"), 0750)
+			if err != nil {
+				panic(err)
+			}
+			err = os.MkdirAll(filepath.Join(context, "abcd"), 0750)
+			if err != nil {
+				panic(err)
+			}
+			_, err = os.Create(filepath.Join(context, "a.txt"))
+			if err != nil {
+				panic(err)
+			}
+
+			helper.DeleteDir(filepath.Join(context, "abcd"))
+
+			helper.ReplaceString(filepath.Join(context, "app", "app.js"), "Hello", "Hello odo")
+		}
+	}()
+
+	success, err := helper.WatchNonRetCmdStdOut(
+		("odo watch --context " + context),
+		time.Duration(10)*time.Minute,
+		func(output string) bool {
+			sourcePath := "/projects/nodejs-web-app"
+
+			stringsMatched := true
+
+			for _, stringToBeMatched := range stringsToBeMatched {
+				if !strings.Contains(output, stringToBeMatched) {
+					stringsMatched = false
+				}
+			}
+
+			if stringsMatched {
+				// Verify delete from component pod
+				stdOut := getContainerExecListdir(runner, platform, cmpName, project, sourcePath)
+				Expect(stdOut).To(ContainSubstring(("a.txt")))
+				Expect(stdOut).To(ContainSubstring((".abc")))
+				Expect(stdOut).To(Not(ContainSubstring(("abcd"))))
+				return true
+			}
+			return false
+		},
+		startSimulationCh,
+		func(output string) bool {
+			return strings.Contains(output, "Waiting for something to change")
+		})
+
+	Expect(success).To(Equal(true))
+	Expect(err).To(BeNil())
+}
+
+func getContainerExecListdir(runner interface{}, platform, cmpName, project, sourcePath string) string {
+	var stdOut string
+	// if !reflect.DeepEqual(runner.(helper.OcRunner), helper.OcRunner{}) {
+
+	if platform == "kube" {
+		ocRunner := runner.(helper.OcRunner)
```

Comment:
we don't need this conversion

Reply:
yep true but since we need it for docker runner; i thought I'd keep the conversion to keep it consistent aesthetically